packages/nuxt/src/components/runtime/client-delayed-component.ts

```diff
@@ -0,0 +1,129 @@
+import { createStaticVNode, defineComponent, getCurrentInstance, h, onBeforeUnmount, onMounted, ref } from 'vue'
+import type { Component, ComponentInternalInstance, Ref } from 'vue'
+// import ClientOnly from '#app/components/client-only'
+import { getFragmentHTML } from '#app/components/utils'
+import { useNuxtApp } from '#app/nuxt'
+import { cancelIdleCallback, requestIdleCallback } from '#app/compat/idle-callback'
+import { onNuxtReady } from '#app'
+import { useIntersectionObserver } from '#app/utils'
+
+function elementIsVisibleInViewport (el: Element) {
+  const { top, left, bottom, right } = el.getBoundingClientRect()
+  const { innerHeight, innerWidth } = window
+  return ((top > 0 && top < innerHeight) ||
+    (bottom > 0 && bottom < innerHeight)) &&
+    ((left > 0 && left < innerWidth) || (right > 0 && right < innerWidth))
+}
+
+/* @__NO_SIDE_EFFECTS__ */
+export const createLazyIOComponent = (componentLoader: Component) => {
+  return defineComponent({
+    inheritAttrs: false,
+    setup (_, { attrs }) {
+      if (import.meta.server) {
+        return () => h(componentLoader, attrs)
+      }
+
+      const nuxt = useNuxtApp()
+      const instance = getCurrentInstance()!
+      const isIntersecting = ref(false)

```

Comment:
slightly misleading, should be `hasIntersected` (it doesn't get toggled to false if it's not in the viewport anymore)

Reply:
Good spot, fixed!